packages/buttons/src/styled/StyledIconButton.ts

```diff
@@ -69,10 +69,10 @@ const iconStyles = (props: IButtonProps & ThemeProps<DefaultTheme>) => {
   `;
 };
 
-export const StyledIconButton = styled(StyledButton as 'button').attrs({
-  'data-garden-id': COMPONENT_ID,
+export const StyledIconButton = styled(StyledButton as 'button').attrs(props => ({
+  'data-garden-id': (props as any)['data-garden-id'] || COMPONENT_ID,
```

Comment:
I dug further into https://github.com/zendeskgarden/react-components/pull/1804 and found some regressions with this method. I'll share my findings and recommendation in that PR. Meanwhile, here's a summary.

Notice how `IconButton` now inherit the `COMPONENT_ID` `buttons.button` from `StyledButton`. 

![Screen Shot 2024-05-15 at 7 37 03 AM](https://github.com/zendeskgarden/react-components/assets/6879688/c2c05ca9-1b6e-45d4-b63b-1532d0077676)

When extending a styled-component with `attrs` already set, the props will include such attributes. In this case `props['data-garden-id'] === 'buttons.button'` and takes precedence over the `StyledIconButton`'s `COMPONENT_ID`.

One way around that is to first check that the `data-garden-id` exist and doesn't match the id from the styled-component being extended:

```suggestion
export const StyledIconButton = styled(StyledButton as 'button').attrs(props => ({
  'data-garden-id': (props as any)['data-garden-id'] && (props as any)['data-garden-id'] !== 'buttons.button' ?
  (props as any)['data-garden-id'] : COMPONENT_ID,
```

I can't recommend this pattern. It's brittle and error-prone. My hunch is that we should set `data-garden-id` on the components we expose to the public and only rely on `attrs` for `data-garden-version` or other immutable attributes.

I'm happy to discuss further.

Reply:
> I can't recommend this pattern.

Thanks for the carefule review. I would say that, within the context of a package, this 2nd pattern (although somewhat awkward) is totally acceptable.

Reply:
The approach I highlighted above is not perfect but it's a good stepping stone.

I [updated all instances](https://github.com/zendeskgarden/react-components/pull/1804/commits/93dfd81e310563b44770202191c1915d70ac2978) in https://github.com/zendeskgarden/react-components/pull/1804. 

How should we move forward with merging these changes into this PR?

Reply:
I've reverted the changes here (and updated PR title, description, and tags) in favor of reviewing #1804